python/ray/util/sgd/v2/trainer.py

```diff
@@ -1,14 +1,30 @@
 import inspect
 import logging
-from typing import Union, Callable, List, TypeVar, Optional, Any, Dict
+from typing import Union, Callable, List, TypeVar, Optional, Any, Dict, Type, \
+    Iterator
 
-from ray.tune import Trainable
 from ray.util.sgd.v2.backends.backend import BackendConfig, BackendExecutor, \
     InactiveWorkerGroupError, SGDBackendError
 from ray.util.sgd.v2.backends.tensorflow import TensorflowConfig
 from ray.util.sgd.v2.backends.torch import TorchConfig
 from ray.util.sgd.v2.callbacks.callback import SGDCallback
 
+# Ray SGD should be usable even if Tune is not installed.
+try:
+    TUNE_INSTALLED = True
+    from ray import tune
+    from ray.tune import Trainable
+    from ray.tune import PlacementGroupFactory
+    from ray.tune.function_runner import wrap_function
+except ImportError:
+    TUNE_INSTALLED = False
+    tune = PlacementGroupFactory = Trainable = object
+
+    def noop():
+        return
+
+    wrap_function = noop
```

Comment:
Are these necessary? What's the difference in behavior if these aren't explicitly set?

Reply:
I don't think there will be any difference in behavior if these aren't set since it will error out anyways if tune isn't installed. But I think it's better to keep these here just in case.